datadog_checks_base/tests/test_agent_check.py

```diff
@@ -376,3 +376,51 @@ def test_metric_limit_instance_config_zero(self, aggregator):
             check.gauge("metric", 0)
         assert len(check.get_warnings()) == 1  # get_warnings resets the array
         assert len(aggregator.metrics("metric")) == 10
+
+
+class TestCheckInitializations:
+    def test_success_only_once(self):
+        class TestCheck(AgentCheck):
+            def __init__(self, *args, **kwargs):
+                super(TestCheck, self).__init__(*args, **kwargs)
+                self.state = 1
+                self.initialize = mock.MagicMock(side_effort=self._initialize)
```

Comment:
```suggestion
                self.initialize = mock.MagicMock(side_effect=self._initialize)
```
?

Reply:
But honestly you don't even need that `self._initialize` method here as the code run it doesn't matter as long as it doesn't fail 🤷‍♂